packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -175,17 +175,17 @@ class VMService {
   Future<Null> get done => _peer.done;
 
   // Events
-  Stream<ServiceEvent> get onDebugEvent => onEvent('Debug');
-  Stream<ServiceEvent> get onExtensionEvent => onEvent('Extension');
+  Future<Stream<ServiceEvent>> get onDebugEvent => onEvent('Debug');
```

Comment:
Use of `Future<Stream<T>>` should never be necessary?

Reply:
Interesting, I didn't know that. Why? What if we are not able to provide `Stream` right away? (which is basically the case here)

Reply:
I may be wrong of course, but I actually think any use of `Future<Stream<T>>` should result in a linter warning. One should always be able to just return a `Stream<T>` immediately and then delay the first event on that stream.

Reply:
Sure, but how do you represent the case where a method that returns `Stream` actually have to `await` for something else? In our case we want to say `(await vmService.onTimelineEvent).listen` to indicate that we don't want continue with rest of `runFromSource` until we get the stream, which happens after vm confirms it received the event subscription.

Reply:
I'd probably contend that the method does two things and should be split.

Reply:
When the method that returns a `Stream` has to await something else, the right solution is to use `async*` and `yield`.

Reply:
I don't think that is the issue here. We want to await an asynchronous stream setup operation before calling runInView. I think it should be done in its own method, separate from the Stream getters

Reply:
Just for cleanliness' sake, I sent https://github.com/flutter/flutter/pull/16711